client/asset/dcr/coin_selection.go

```diff
@@ -0,0 +1,210 @@
+		return nil
+	}
+	// if i == len(utxos)-1 { return utxos[i:] } // shouldn't happen if each in set are < amt
+
+	// Find the last one to meet amt, as small as possible.
+	rem, set := utxos[:i+1], utxos[i+1:]
+	idx := sort.Search(len(rem), func(i int) bool {
+		return sum+toAtoms(rem[i].rpc.Amount) >= amt
+	})
+
+	return append([]*compositeUTXO{rem[idx]}, set...)
+}
+
+// subsetSmallBias begins by summing from the smallest UTXO up until the sum is
+// at least the requested amount. Then it drops the smallest ones it had
+// selected if they are not required to reach the amount.
+func subsetSmallBias(amt uint64, utxos []*compositeUTXO) []*compositeUTXO {
+	// Add from smallest up until sum is enough.
+	var sum uint64
+	var idx int
+	for i, utxo := range utxos {
+		sum += toAtoms(utxo.rpc.Amount)
+		if sum >= amt {
+			idx = i
+			break
+		}
+	}
+	if sum < amt {
+		return nil
+	}
+	set := utxos[:idx+1]
+
+	// Now drop excess small ones.
+	for i, utxo := range set {
+		sum -= toAtoms(utxo.rpc.Amount)
+		if sum < amt {
+			idx = i // needed this one
+			break
+		}
+	}
+	return set[idx:]
+}
+
+// leastOverFund attempts to pick a subset of the provided UTXOs to reach the
+// required amount with the objective of minimizing the total amount of the
+// selected UTXOs. This is different from the objective used when funding
+// orders, which is to minimize the number of UTXOs (to minimize fees).
+//
+// The UTXOs MUST be sorted in ascending order (smallest first, largest last)!
+//
+// This begins by partitioning the slice before the smallest single UTXO that is
+// large enough to fully fund the requested amount, if it exists. If the smaller
+// set is insufficient, the single largest UTXO is returned. If instead the set
+// of smaller UTXOs has enough total value, it will search for a subset that
+// reaches the amount with least over-funding (see subsetSmallBias and
+// subsetLargeBias). If that subset has less combined value than the single
+// sufficiently-large UTXO (if it exists), the subset will be returned,
+// otherwise the single UTXO will be returned.
+//
+// If the provided UTXO set has less combined value than the requested amount a
+// nil slice is returned.
+func leastOverFund(amt uint64, utxos []*compositeUTXO) []*compositeUTXO {
+	if amt == 0 || sumUTXOs(utxos) < amt {
+		return nil
+	}
+
+	// Partition - smallest UTXO that is large enough to fully fund, and the set
+	// of smaller ones.
+	idx := sort.Search(len(utxos), func(i int) bool {
+		return toAtoms(utxos[i].rpc.Amount) >= amt
+	})
+	var small []*compositeUTXO
+	var single *compositeUTXO // only return this if smaller ones would use more
+	if idx == len(utxos) {    // no one is enough
+		small = utxos
+	} else {
+		small = utxos[:idx]
+		single = utxos[idx]
+	}
+
+	// Find a subset of the small UTXO set with smallest combined amount.
+	var set []*compositeUTXO
+	if sumUTXOs(small) >= amt {
+		set = subsetLargeBias(amt, small)
```

Comment:
Was on my todo list.  :blush: Just threw together a few heiristics, but you have a point that it can be solved optimally.  Seems akin to the fractional knapsack problem (reversed).  Will give it a shot.